hoot-core/src/main/cpp/hoot/core/io/OsmApiWriter.cpp

```diff
@@ -265,9 +265,20 @@ void OsmApiWriter::_changesetThreadFunc()
         //  Split the changeset on conflict errors
         switch (info->status)
         {
+        case 409:   //  Conflict, check for version conflicts and fix, or split and continue
+          {
+            if (_fixConflict(request, workInfo, info->response))
+            {
+              _workQueueMutex.lock();
+              _workQueue.push(workInfo);
+              _workQueueMutex.unlock();
+              //  Loop back around to work on the next changeset
+              continue;
+            }
+            //  Fall through here to split the changeset and retry
+          }
```

Comment:
Version conflicts are dealt with here and retried after being fixed.  If they aren't able to be fixed, then the changeset is split and tried again.